crates/genesis/src/system.rs

```diff
@@ -453,6 +498,54 @@ impl core::fmt::Display for GasLimitUpdateError {
 
 impl core::error::Error for GasLimitUpdateError {}
 
+/// An error for updating the EIP-1559 parameters on the [SystemConfig].
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub enum EIP1559UpdateError {
+    /// Invalid data length.
+    InvalidDataLen(usize),
+    /// Failed to decode the data pointer argument from the eip 1559 update log.
+    PointerDecodingError,
+    /// The data pointer is invalid.
+    InvalidDataPointer(u64),
+    /// Failed to decode the data length argument from the eip 1559 update log.
+    LengthDecodingError,
+    /// The data length is invalid.
+    InvalidDataLength(u64),
+    /// Failed to decode the gas limit argument from the eip 1559 update log.
```

Comment:
I think this is bad copy pasta - should be
```suggestion
    /// Failed to decode the eip1559 params argument from the eip 1559 update log.
```